stanford-earth/SE3_BLT
# Earth 408 https redirect

# READY FOR REVIEW

# Summary
In the "post-drupal-scaffold-cmd" of the scripts section of composer.json, does:
- Patches .htaccess file to force redirect of any URL beginning http://earthdev.stanford.edu to https
- Patches .htaccess file to allow URLS for simplesaml
- Runs "blt simplesamlphp:init" which adds simplesamlphp to the vendors directory and creates a symlink from docroot.

Note 1: this can be run without concern on local, dev, and stage sites without affecting their operation. Redirect only refers to specific host earthdev.stanford.edu and simplesaml init and symlink can be ignored.

Note 2: if "composer drupal-scaffold" is re-run, it should detect that simplesamlphp is already installed and do nothing; the symlink in docroot will be recreated if needed.

# Needed By (Date)
- Next release.

# Urgency
- none

# Steps to Test

1. cd into your local installation (/WWW/se3_blt, for example)
2. git fetch; git checkout EARTH-408-https-redirect
3. run "composer drupal-scaffold"
4. a simplesamlphp directory should now be under <project>/vendor
5. a simplesaml symlink to <project>/vendor/simplesamlphp/www should now be in docroot
6. externalauth and simplesamlphp_auth modules should now be in <project>/docroot/modules/contrib, if they weren't already there (see EARTH-409)
7. rm simplesaml symlink from docroot
8. re-run "composer drupal-scaffold"; the symlink should be back.
9. feel free to manually modify the "earthdev.stanford.edu" RewriteRule in .htaccess to something that has https in order to test.

# Affected versions or modules

# Associated Issues and/or People
EARTH-409 installs some additional simplesamlphp_auth module configuration in Drupal for the production site.
